selftests/functional/test_utils.py

```diff
@@ -155,5 +160,35 @@ def test_process_run(self):
     def tearDown(self):
         shutil.rmtree(self.base_logdir)
 
+
+def file_lock_action(args):
+    path, players = args
+    max_individual_timeout = 0.021
+    max_timeout = max_individual_timeout * players
+    with FileLock(path, max_timeout):
+        sleeptime = random.random() / 100
+        time.sleep(sleeptime)
+
+
+class FileLockTest(unittest.TestCase):
+
+    def setUp(self):
+        self.tmpdir = tempfile.mkdtemp(prefix='avocado_' + __name__)
+
+    @unittest.skipIf(os.environ.get("AVOCADO_CHECK_LONG") != "1",
+                     "Skipping test that takes a long time to run")
+    def test_filelock(self):
+        players = 1000
+        pool = multiprocessing.Pool(players)
+        args = [(self.tmpdir, players)] * players
+        try:
+            pool.map(file_lock_action, args)
+        except:
+            self.fail('Failed to run FileLock with %s players' % players)
```

Comment:
Actually one more think. I'd be nice (if not essential) to see the failure. I'd simply use `print "Running with %s players"` and then I'd remove the `try/except`. That way you get the original exception as well as the output and in case there is no output unittest hides the stdout.